checks/database_backup_gcp.go

```diff
@@ -0,0 +1,56 @@
+package checks
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/flanksource/canary-checker/api/context"
+	v1 "github.com/flanksource/canary-checker/api/v1"
+	"github.com/flanksource/canary-checker/pkg"
+	"github.com/flanksource/canary-checker/pkg/clients/gcp"
+)
+
+func GCPDatabaseBackupCheck(ctx *context.Context, check v1.DatabaseBackupCheck) pkg.Results {
+	databaseScanObjectCount.WithLabelValues(check.GCP.Project, check.GCP.Instance).Inc()
+	result := pkg.Success(check, ctx.Canary)
+	var results pkg.Results
+	results = append(results, result)
+
+	svc, err := gcp.NewSQLAdmin(ctx, *check.GCP.GCPConnection)
+	if err != nil {
+		databaseScanFailCount.WithLabelValues(check.GCP.Project, check.GCP.Instance).Inc()
+		return results.ErrorMessage(err)
+	}
+	// Only checking one backup for now, but setting up the logic that this could maybe be configurable.
+	// Would need some extra parsing on the age to select latest
+	backupList, err := svc.BackupRuns.List(check.GCP.Project, check.GCP.Instance).MaxResults(1).Do()
+	if err != nil {
+		databaseScanFailCount.WithLabelValues(check.GCP.Project, check.GCP.Instance).Inc()
+		return results.ErrorMessage(err)
+	}
+	var errorMessages []string
+	for _, backup := range backupList.Items {
+		if backup.Status != "SUCCESSFUL" {
+			errorMessages = append(errorMessages, backup.Error.Message)
+		}
+	}
+	if check.MaxAge > 0 {
```

Comment:
Can MaxAge of `0` be a valid case?
In that case I would also make it a pointer and have it as `check.MaxAge != nil`

Reply:
I don't think it would be a reasonable case to always expect the most recent backup to be under a minute old...The most frequent backup schedule we would currently be monitoring is daily.